Unable to update from About

### Steps to reproduce

1. Go to settings
2. Tap `About`
3. Click yes on new update notification

### Expected behavior

The App should update

### Actual behavior

The App cannot update on its own

### LibreTube version

0.7.0

### Android version

Android 11

### Other details

java.lang.RuntimeException: Unable to start service com.github.libretube.services.UpdateService@7d9842d with Intent { cmp=com.github.libretube/.services.UpdateService (has extras) }: java.lang.SecurityException: No permission to write to file:///storage/emulated/0/Download/release.apk: Neither user 10161 nor current process has android.permission.WRITE_EXTERNAL_STORAGE.
 at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4017)
 at android.app.ActivityThread.access$1800(ActivityThread.java:233)
 at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1912)
 at android.os.Handler.dispatchMessage(Handler.java:106)
 at android.os.Looper.loop(Looper.java:233)
 at android.app.ActivityThread.main(ActivityThread.java:7225)
 at java.lang.reflect.Method.invoke(Native Method)
 at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:499)
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:962)
Caused by: java.lang.SecurityException: No permission to write to file:///storage/emulated/0/Download/release.apk: Neither user 10161 nor current process has android.permission.WRITE_EXTERNAL_STORAGE.
 at android.os.Parcel.createException(Parcel.java:1966)
 at android.os.Parcel.readException(Parcel.java:1934)
 at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:183)
 at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:135)
 at android.content.ContentProviderProxy.insert(ContentProviderNative.java:476)
 at android.content.ContentResolver.insert(ContentResolver.java:1604)
 at android.app.DownloadManager.enqueue(DownloadManager.java:1028)
 at com.github.libretube.services.UpdateService.onStartCommand(UpdateService.kt:106)
 at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3984)
 ... 8 more

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

> ### Steps to reproduce
> 
>     1. Go to settings
> 
>     2. Tap `About`
> 
>     3. Click yes on new update notification
> 
> 
> ### Expected behavior
> 
> The App should update
> ### Actual behavior
> 
> The App cannot update on its own
> ### LibreTube version
> 
> 0.7.0
> ### Android version
> 
> Android 11
> ### Other details
> 
> java.lang.RuntimeException: Unable to start service com.github.libretube.services.UpdateService@7d9842d with Intent { cmp=com.github.libretube/.services.UpdateService (has extras) }: java.lang.SecurityException: No permission to write to file:///storage/emulated/0/Download/release.apk: Neither user 10161 nor current process has android.permission.WRITE_EXTERNAL_STORAGE. at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4017) at android.app.ActivityThread.access$1800(ActivityThread.java:233) at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1912) at android.os.Handler.dispatchMessage(Handler.java:106) at android.os.Looper.loop(Looper.java:233) at android.app.ActivityThread.main(ActivityThread.java:7225) at java.lang.reflect.Method.invoke(Native Method) at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:499) at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:962) Caused by: java.lang.SecurityException: No permission to write to file:///storage/emulated/0/Download/release.apk: Neither user 10161 nor current process has android.permission.WRITE_EXTERNAL_STORAGE. at android.os.Parcel.createException(Parcel.java:1966) at android.os.Parcel.readException(Parcel.java:1934) at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:183) at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:135) at android.content.ContentProviderProxy.insert(ContentProviderNative.java:476) at android.content.ContentResolver.insert(ContentResolver.java:1604) at android.app.DownloadManager.enqueue(DownloadManager.java:1028) at com.github.libretube.services.UpdateService.onStartCommand(UpdateService.kt:106) at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3984) ... 8 more
> ### Acknowledgements
> 
>     * [x]  I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
> 
>     * [x]  I have written a short but informative title.
> 
>     * [x]  I will fill out all of the requested information in this form.

Libretube doesn't have storage permission in your device. 🙇‍♂️

--- a/app/src/main/java/com/github/libretube/services/UpdateService.kt
+++ b/app/src/main/java/com/github/libretube/services/UpdateService.kt
@@ -11,6 +11,7 @@ import android.os.Environment
 import android.os.IBinder
 import android.widget.Toast
 import com.github.libretube.R
+import com.github.libretube.util.DownloadHelper
 import java.io.File
 
 class UpdateService : Service() {
@@ -28,9 +29,7 @@ class UpdateService : Service() {
     }
 
     private fun downloadApk(downloadUrl: String) {
-        val dir = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS)
-        // val dir = applicationContext.getExternalFilesDir(Environment.DIRECTORY_DOWNLOADS)
-        file = File(dir, "release.apk")
+        file = File(getDownloadDirectory(), "release.apk")
 
         val request: DownloadManager.Request =
             DownloadManager.Request(Uri.parse(downloadUrl))
@@ -78,6 +77,12 @@ class UpdateService : Service() {
                 }
             }
         }
+    }
+
+    private fun getDownloadDirectory(): File {
+        val downloadsDir = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS)
+        if (!downloadsDir.canWrite()) return DownloadHelper.getOfflineStorageDir(this)
+        return downloadsDir
     }
 
     override fun onDestroy() {

--- a/app/src/main/java/com/github/libretube/util/DownloadHelper.kt
+++ b/app/src/main/java/com/github/libretube/util/DownloadHelper.kt
@@ -11,7 +11,7 @@ object DownloadHelper {
     const val METADATA_DIR = "metadata"
     const val THUMBNAIL_DIR = "thumbnail"
 
-    private fun getOfflineStorageDir(context: Context): File {
+    fun getOfflineStorageDir(context: Context): File {
         if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) return context.filesDir
 
         return try {

